enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,1054 @@
+     cluster
+3. aws-ebs-csi-driver-operator reports its status in ClusterCSDriver.Status in the guest cluster.
+4. cluster-storage-operator collects ClusterCSDriver.Status and translates it to its own ClusterOperator.Status in the
+   guest cluster.
+
+All storage operators (cluster-storage-operator, csi-driver-operator, and csi-snapshot-controller-operator) behave the
+same as in standalone OCP, they just create / watch / sync objects in two clusters instead of one. This allows us to
+reuse the existing operators.
+
+<!--
+Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.
+
+Use sub-sections to explain variations, such as for error handling,
+failure recovery, or alternative outcomes.
+
+For example:
+
+**cluster creator** is a human user responsible for deploying a
+cluster.
+
+**application administrator** is a human user responsible for
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials.
+
+-->
+
+#### Variation [optional]
+
+N/A ?
+
+<!--
+If the cluster creator uses a standing desk, in step 1 above they can
+stand instead of sitting down.
+
+See
+https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md#high-level-end-to-end-workflow
+and https://github.com/openshift/enhancements/blob/master/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md for more detailed examples.
+
+-->
+
+### API Extensions
+
+No API changes for now.
+
+cluster-storage-operator, cluster-csi-snapshot-controller-operator and aws-ebs-csi-driver-operator currently allow users
+only to set ManagementState, LogLevel and OperatorLogLevel in their CRs (`Storage`, `CSISnapshotController`
+and `ClusterCSIDriver`). Since hosted-cluster-config-operator will overwrite any user changes in these CRs (see below),
+users, SREs, or developers are not able to set log level of storage components directly, they have to pause
+hosted-cluster-config-operator first.
+
+In a future HyperShift release, log level could be set as a field in HostedCluster / HostedControlPlane and then
+projected into storage operators. Such API is out of scope of this enhancement.
+
+In a future OCP release, CSI driver operators may have more configuration fields, such as
+[a flag that disables StorageClass creation](https://github.com/openshift/enhancements/pull/1152/files). How user sets
+them is out of scope of this enhancement.
```

Comment:
Currently, library-go's config observation controllers mutate the ClusterCSIDriver.Spec. We use that for observing the proxy settings.

This means that if any other operator mutates the ClusterCSIDriver object, it will race with the AWS CSI Driver Operator.

We may address this by disabling the proxy observation in Hypershift or reworking it to not mutate the ClusterCSIDriver.Spec.

Reply:
Added a note about it to "aws-ebs-csi-driver-operator inputs" and "Required changes for AWS EBS CSI driver in HyperShift" / hosted-cluster-config-operator.